tests/vxlan/test_vxlan_ecmp.py

```diff
@@ -621,13 +661,515 @@ def dump_self_info_and_run_ptf(self, tcname, encap_type, expect_encap_success):
+            logger.info("Clear out the mappings, and keep only the deleted dest and nhs.")
+            self.setup[encap_type]['dest_to_nh_map'][vnet] = {}
+            self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest] = tc10_nhs
+
+            logger.info("The deleted route should fail to receive traffic.")
+            self.dump_self_info_and_run_ptf("tc10", encap_type, False)
+
+            # all others should be working.
+            # Housekeeping:
+            logger.info("Restore the mapping of dest->nhs.")
+            self.setup[encap_type]['dest_to_nh_map'] = dict(full_map)
+            logger.info("Remove the deleted entry alone.")
+            del self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest]
+            del_needed = False
+
+            logger.info("Check the traffic is working in the other routes.")
+            self.dump_self_info_and_run_ptf("tc10", encap_type, True)
+
+        except:
+            self.setup[encap_type]['dest_to_nh_map'] = dict(full_map)
+            logger.info("Remove the deleted entry alone.")
+            if del_needed:
+                del self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest]
+            raise
+
+class Test_VxLAN_ecmp_random_hash(Test_VxLAN):
+    def test_vxlan_random_hash(self, setUp, encap_type):
+        '''
+            tc11: set tunnel route 3 to endpoint group c = {c1, c2, c3}. ensure c1, c2, and c3 matches to underlay default route. send 1000 pkt with random hash to route 3's prefix dst.
+        '''
+        self.setup = setUp
+
+        logger.info("Chose a vnet for testing.")
+        vnet = self.setup[encap_type]['vnet_vni_map'].keys()[0]
+
+        logger.info("Create a new destination and 3 nhs for it.")
+        tc11_new_dest = get_ip_address(af=get_payload_version(encap_type), netid=DESTINATION_PREFIX)
+        tc11_new_nhs = []
+        for i in range(3):
+            tc11_new_nhs.append(get_ip_address(af=get_outer_layer_version(encap_type), netid=NEXTHOP_PREFIX))
+
+        # the topology always provides the default routes for any ip address.
+        # so it is already taken care of.
+
+        logger.info("Map the new dest and nhs.")
+        tc11_config = '[\n' + create_single_route(vnet, tc11_new_dest, HOST_MASK[get_payload_version(encap_type)], tc11_new_nhs, "SET") + '\n]'
+        self.setup[encap_type]['dest_to_nh_map'][vnet][tc11_new_dest] = tc11_new_nhs
+
+        logger.info("Apply the config in the DUT and verify traffic. The random hash and ECMP check is already taken care of in the VxLAN PTF script.")
+        apply_config_in_swss(self.setup['duthost'], tc11_config, "vnet_route_tc11_"+encap_type)
+        self.dump_self_info_and_run_ptf("tc11", encap_type, True, packet_count=1000)
+
+class Test_VxLAN_underlay_ecmp(Test_VxLAN):
+    @pytest.mark.parametrize("ecmp_path_count", [1, 2])
+    def test_vxlan_modify_underlay_default(self, setUp, minigraph_facts, encap_type, ecmp_path_count):
```

Comment:
We are facing problems when running this test. When ecmp_path_count=1, it chooses one portchannel. and shuts down all other portchannels. However this includes the portchannel that is selected as t0 which results in no traffic ingressing the DUT. Below is a dump from PDB. The selected interface is PortChannel1014 which is down because it is part of T2 list too (you can see it is mapped to ports 8 and 9). So the same portchannel is part of T0 ports as well as T2 ports. Shouldn't they be mutually exclusive? Can you please check the logic which gets the T0 ports? https://github.com/Azure/sonic-mgmt/blob/af025eca0a87f4dc61267046a06de2303031f2e2/tests/vxlan/test_vxlan_ecmp.py#L130
(Pdb) self.setup[encap_type]['selected_interfaces']
[u'PortChannel1014']
(Pdb) self.setup[encap_type]['t2_ports']
[0, 1, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
(Pdb)

Reply:
@dgsudharsan , are you saying the selected portchannel is shut down? I thought you mentioned, "When ecmp_path_count=1, it chooses one portchannel. and shuts down all other portchannels"

Reply:
@prsunny I believe there needs to be a portchannel from T0 to T1 and one from T1 to T2. However the test shuts all portchannels except one T1 to T2.
The packets from T0(PTF) are not reaching the DUT (T1). This is due to the fact that the port channel allocated for T0-T1 is also present in the list of portchannels between T1 and T2.

Reply:
@rraghav-cisco , can you please take a look at this comment?

Reply:
Looking into this. I am surprised what causes this issue. I will check the logs.

Reply:
This is the long from the tests. 
As you can see PortChannel1014 is shut but its also the portchannel the t0 choses to send non encapsulated packet.

 Dumping T2 link info: [u'PortChannel105', u'PortChannel102', u'PortChannel1011', u'PortChannel1020', u'PortChannel1014', u'PortChannel1017', u'PortChannel1023', u'PortChannel108']
 Reverse the action: bring up the selected_intfs and shutdown others.
 Shutdown other interfaces.
 tests.ptf_runner:ptf_runner.py:54 Exception caught while executing case: vxlan_traffic.VXLAN. Error message: Traceback (most recent call last):

Reply:
I think I might have found the problem. Working on a fix and checking it.